### PR TITLE
Visx bid adapter: import utilities and retrieve data from user on ortb2

### PR DESCRIPTION
**Pull request by Goncakkd**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Retrieved and sent user data (for getting topics data)

### CODE REVIEW

**Comment by Goncakkd** [comment]
> Whoa there, partner! 🌵🤠 We wrangled some duplicated code in your PR:
> 
> * [`modules/luponmediaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/luponmediaBidAdapter.js#L277-L283) has 8 duplicated lines with [`modules/visxBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/visxBidAdapter.js#L277-L283)
> * [`modules/gridBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/gridBidAdapter.js#L513-L525) has 14 duplicated lines with [`modules/visxBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/visxBidAdapter.js#L321-L333)
> * [`modules/carodaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/carodaBidAdapter.js#L199-L204) has 7 duplicated lines with [`modules/visxBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/visxBidAdapter.js#L277-L283)
> 
> Reducing code duplication by importing common functions from a library not only makes our code cleaner but also easier to maintain. Please move the common code from both files into a library and import it in each. Keep up the great work! 🚀



> Whoa there, partner! 🌵🤠 We wrangled some duplicated code in your PR:
> 
> * [`modules/luponmediaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/luponmediaBidAdapter.js#L277-L283) has 8 duplicated lines with [`modules/visxBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/visxBidAdapter.js#L277-L283)
> * [`modules/gridBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/gridBidAdapter.js#L513-L525) has 14 duplicated lines with [`modules/visxBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/visxBidAdapter.js#L321-L333)
> * [`modules/carodaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/carodaBidAdapter.js#L199-L204) has 7 duplicated lines with [`modules/visxBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/1bf5cb47749bf421300ded951801d132dc952f3e/modules/visxBidAdapter.js#L277-L283)
> 
> Reducing code duplication by importing common functions from a library not only makes our code cleaner but also easier to maintain. Please move the common code from both files into a library and import it in each. Keep up the great work! 🚀

Hello, check for duplicated code prevents us to merge it. Touching the code of the other bid adapters doesnot make sense. Can you please help us?

File: modules/visxBidAdapter.js [hunk 1]
```diff
@@ -116,6 +117,24 @@ export const spec = {
...
+      if (payloadUserEids || payload.gdpr_consent || vads) {
+        user = {
+          ext: {
+            ...(payloadUserEids && { eids: payloadUserEids }),
+            ...(payload.gdpr_consent && { consent: payload.gdpr_consent }),
+            ...(vads && { vads })
+          }
+        };
+      }
+      if (user) {
```
Comment (patmmccann) [inline modules/visxBidAdapter.js:133]: please cover this behavior in a test so you can be confident it will not break. Thanks!

**Review by patmmccann — CHANGES_REQUESTED** [review]
Please add one test and use /libraries instead of utils for the moved function. Thanks!

**Comment by patmmccann** [comment]
> * [`modules/gothamadsBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/gothamadsBidAdapter.js#L304-L311) has 9 duplicated lines with [`modules/luponmediaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/luponmediaBidAdapter.js#L543-L551)
> * [`modules/distroscaleBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/distroscaleBidAdapter.js#L251-L256) has 7 duplicated lines with [`modules/luponmediaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/luponmediaBidAdapter.js#L159-L164)
> * [`modules/carodaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/carodaBidAdapter.js#L158-L173) has 17 duplicated lines with [`modules/dianomiBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/dianomiBidAdapter.js#L86-L104)
> * [`modules/adtrueBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/adtrueBidAdapter.js#L551-L562) has 13 duplicated lines with [`modules/luponmediaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/luponmediaBidAdapter.js#L159-L170)
> 
Feel free to ignore non visx issues!

**Comment by Goncakkd** [comment]
> > * [`modules/gothamadsBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/gothamadsBidAdapter.js#L304-L311) has 9 duplicated lines with [`modules/luponmediaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/luponmediaBidAdapter.js#L543-L551)
> > * [`modules/distroscaleBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/distroscaleBidAdapter.js#L251-L256) has 7 duplicated lines with [`modules/luponmediaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/luponmediaBidAdapter.js#L159-L164)
> > * [`modules/carodaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/carodaBidAdapter.js#L158-L173) has 17 duplicated lines with [`modules/dianomiBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/dianomiBidAdapter.js#L86-L104)
> > * [`modules/adtrueBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/adtrueBidAdapter.js#L551-L562) has 13 duplicated lines with [`modules/luponmediaBidAdapter.js`](https://github.com/prebid/Prebid.js/blob/c1bb11fddc02ebc66559d741eaa26224e73b75b7/modules/luponmediaBidAdapter.js#L159-L170)
> 
> Feel free to ignore non visx issues!

How can i ignore them? Merging is blocked because of this.

**Comment by patmmccann** [comment]
I'm in charge of the merge, I can override

**Review by patmmccann — APPROVED** [review]

**Pull request merged**